lib/libc/glibc/README.md

```diff
@@ -0,0 +1,85 @@
+# Zig GNU C Library ("glibc") Support
+
+*Date*: October, 2023
+
+Zig supports building binaries that will dynamically link against the
+[GNU C Library ("glibc")](https://www.gnu.org/software/libc/) when run.
+This support extends across a range of glibc versions.
+
+By default, Zig binaries will not depend on any external C library, but
+they can be linked against one with the `-lc` option.  The compilation
+target defines which C library: `musl` is the statically linked
+[musl C library](https://musl.libc.org/) or `gnu` which will be a
+run-time, dynamic linking dependency on glibc.
+
+A specific C library version can be chosen with an appropriate `-target`.
+For example, `-target native-native-gnu.2.19` will use the default CPU and
+OS targets, but will link in a run-time dependency on glibc v2.19 (or
+later).  Use `zig env` to show the default targets and versions.
+
+Glibc symbols are defined in the `std.c.` namespace in Zig, though the
+`std.os.` namespace is generally what should be used to access C-library
+APIs in Zig code (it is defined depending on the linked C library).
+
+See `src/glibc.zig` for how Zig will build the glibc components.  The core
+C library is sufficient only for compile-time linking (it is a stub
+library that just needs to indicate that a given symbol will be present at
+run-time, it doesn't actually have to implement said symbol).
+
+## Targets
+
+The GNU C Library supports a very wide set of platforms and architectures.
+The current Zig support for glibc only supports Linux.  Extend
+`src/glibc.zig` to support additional platforms.
+
+Zig supports glibc versions back to v2.17 (2012) as the Zig standard
+library depends on symbols that were introduced in 2.17.
+
+## Glibc stubs
+
+The file `lib/libc/glibc/abilist` is a Zig-specific binary blob that
+defines the supported glibc versions and the set of symbols each version
+must define.  See https://github.com/ziglang/glibc-abi-tool for the
+tooling to generate this blob.  The code in `glibc.zig` parses the abilist
+to build version-specific stub libraries on demand.
+
+The generated stub library is used for compile-time linking, with the
+expectation that at run-time the real glibc library will provide the
+actual symbol implementations.
+
+### Public Headers
+
+The glibc headers are in `lib/libc/include/generic-glibc/`.  These are
+customized and have a couple Zig-specific `#ifdef`s to make the single set
+of headers represent any of the supported glibc versions.  There are
+currently a handful of patches to these headers to represent new features
+(e.g. `reallocarray`) or changes in implementation (e.g., the `stat()`
+family of functions).
+
```

Comment:
mention the https://github.com/ziglang/universal-headers/ project. I am likely going to merge https://github.com/ziglang/universal-headers/pull/7 shortly.

edit: you have a mention below that should be moved up here

Reply:
Moved universal-headers reference up here.